### ISSUE TITLE
UX Concern

### ISSUE THREAD

**Issue by Lewiscowles1986** [cmpolis/smart-table-scroll#3]
Does it not trouble you that reading 1 million, or even 10,000 rows will be an extreme waste of time for any human?

For my own business software I begrudgingly added infinite scroll for people that want to view "all data", and simply say I am not responsible for browser limitations. I think it's amazing you overcame these, in a method that if I understand it, ensures that records displayed are limited, keeping speed and responsiveness, and only the visible records + buffer, with a known finite size number of display elements; are overwritten based upon scroll position with some CSS magic.

I would still like it if you made it absolutely clear to everyone, that scrolling through even 1,000 rows to manually review is going to take longer than using the advanced multi-faceted search and reporting capabilities of modern software. 

While computers are great at handling 1 million of anything, people are really not, and benefit from silo'ing such figures to comprehensible groups; so while I think this is amazing, I would like to request that a statement about basic common sense for end-user experience and productivity be added to this. 

Basically you should never need to display or access an entire data-set of 1 million things; to a human (their brain is fundamentally incapable of storing them all without a long period of time, that is incompatible with staring at a screen, or browsing a web-page).

**Comment by cmpolis** [comment]
Hi Lewis,

Thanks for your interest in this project and opening this discussion! I'll be brief since this is very much a philosophical discussion and not a technical one.

Yes, manually going through a million or ten thousand unsorted rows makes for bad ux. This tool is intended to be used in conjunction with searching/filtering/ordering functionality_(eg: [project that is using smart-table-scroll](https://github.com/cmpolis/datacomb))_. That being said, some people do have business processes that require exploring larger datasets and this is a tool to help in building tools for those processes; I built this with a few concrete use cases in mind.

The demo and notes use `1 million` to show that this will scale to _virtually_ any `n` rows
....and for dramatic effect :smile:

**Comment by Lewiscowles1986** [comment]
Fantastic! :+1: 

I Really just wanted to check at the source that this was not the intended usage, thanks for clearing up :smile:
